<a href="https://colab.research.google.com/github/Nandinikhandade/MachineLearning/blob/main/CaseStudy_Logit.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [88]:
import numpy as np
import pandas as pd
import os
os.getcwd()     #get current working directory
os.chdir("/content/drive/MyDrive")
os.getcwd

import statsmodels
from statsmodels import stats
from statsmodels.stats import weightstats as ssw

import statsmodels.api as sm
from statsmodels.formula.api import ols



In [89]:
df=pd.read_excel('CaseStudy.xlsx')
df.head()

,Age,Income,Gender,Region,Target
0,58,51722,Female,South,1
1,48,98373,Male,South,1
2,34,51536,Male,South,1
3,62,62768,Male,South,1
4,27,83132,Female,South,0


In [90]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Age     3000 non-null   int64 
 1   Income  3000 non-null   int64 
 2   Gender  3000 non-null   object
 3   Region  3000 non-null   object
 4   Target  3000 non-null   int64 
dtypes: int64(3), object(2)
memory usage: 117.3+ KB


In [91]:
df = pd.get_dummies(df,columns=["Region","Gender"],drop_first=True).astype(int)

In [92]:
df.columns

Index(['Age', 'Income', 'Target', 'Region_South', 'Region_West',
       'Gender_Male'],
      dtype='object')

In [93]:
df.head()

,Age,Income,Target,Region_South,Region_West,Gender_Male
0,58,51722,1,1,0,0
1,48,98373,1,1,0,1
2,34,51536,1,1,0,1
3,62,62768,1,1,0,1
4,27,83132,0,1,0,0


In [94]:
X = df.drop("Target", axis=1)

y = df["Target"]

In [95]:
X.head()

,Age,Income,Region_South,Region_West,Gender_Male
0,58,51722,1,0,0
1,48,98373,1,0,1
2,34,51536,1,0,1
3,62,62768,1,0,1
4,27,83132,1,0,0


In [96]:
X_train, X_test, y_train, y_test = train_test_split( X, y,test_size=0.20,random_state=42)


In [97]:
X_train=sm.add_constant(X_train,prepend=False)
X_test=sm.add_constant(X_test,prepend=False)

In [98]:
X_train.head()

,Age,Income,Region_South,Region_West,Gender_Male,const
642,31,70093,0,0,1,1.0
700,57,36028,0,0,1,1.0
226,63,64980,1,0,0,1.0
1697,66,73405,1,0,1,1.0
1010,50,68559,1,0,1,1.0


In [99]:
X_train.shape

(2400, 6)

In [100]:
X_test.shape

(600, 6)

In [101]:
model1=sm.Logit(y_train,X_train).fit()
print(model1.summary())

Optimization terminated successfully.
         Current function value: 0.458699
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:                 Target   No. Observations:                 2400
Model:                          Logit   Df Residuals:                     2394
Method:                           MLE   Df Model:                            5
Date:                Tue, 23 Jun 2026   Pseudo R-squ.:                  0.1572
Time:                        17:23:16   Log-Likelihood:                -1100.9
converged:                       True   LL-Null:                       -1306.2
Covariance Type:            nonrobust   LLR p-value:                 1.464e-86
                   coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
Age              0.0529      0.004     13.099      0.000       0.045       0.061
Income        -3.31e-06

z=log of odds ratio
1. H0 = age does not influence target
2. checking p value if p value less than level of significance
3. we reject the null hypothesis.
4. it means age does influence the target
5. for every unit increase in age the log of odds ratio in favor of getting target change by 0.0529



linear
5. for every unit increase in age the target will increase by 0.059

categorical regression
linear regresssion ..
multiple linear

region
1.h0= with respect to female if a male is selected that it does not influence the target

2. checking p value if p value less than level of significance

3. we reject the null hypothesis

4. it means with respect to female if a male is selcted that it does influence the target

5. choosing a Male instead of a Female increases the target variable directly by 0.8082 units. This effect is statistically highly significant ((p < 0.001)).